openscap_report/report_generators/html_templates/js/oval_graph_generation_script.js

```diff
@@ -532,6 +530,57 @@ function get_OVAL_test_info(test_info) {
     }
     table.appendChild(get_table_header(objects));
     table.appendChild(get_table_body(objects));
+}
+
+function get_OVAL_state_labels(oval_state) {
+    const div = DIV.cloneNode();
+    const text = P.cloneNode();
+    text.textContent ='This is expected state of OVAL object: ';
+    text.appendChild(BR.cloneNode());
+
+    div.appendChild(text);
+    div.appendChild(get_label("", oval_state.state_id));
+    div.appendChild(get_label("pf-m-blue", oval_state.comment));
+    return div;
+}
+
+function generate_OVAL_state(test_info, div) {
+    if (test_info.oval_state === null) {
+        return;
+    }
+    div.appendChild(BR.cloneNode());
+    div.appendChild(get_OVAL_state_labels(test_info.oval_state));
+    const table_div = DIV.cloneNode();
+    table_div.className = "pf-c-scroll-inner-wrapper oval-test-detail-table";
+    div.appendChild(table_div);
+    const table = TABLE.cloneNode();
+    table.className = "pf-c-table pf-m-compact pf-m-grid-md";
+    table.setAttribute("role", "grid");
+    table_div.appendChild(table);
+
+    const objects = [];
+    for (const data of test_info.oval_state.state_data) {
+        objects.push(filter_object(data, test_info.oval_state));
+    }
+    table.appendChild(get_table_header(objects));
+    table.appendChild(get_table_body(objects));
+}
+
+
+function get_OVAL_test_info(test_info) {
+    const div = DIV.cloneNode();
+    div.className = "pf-c-accordion__expanded-content-body";
+    div.appendChild(get_label("", test_info.test_id));
+    div.appendChild(get_label("pf-m-blue", test_info.comment));
```

Comment:
The comment becames a tooltip  also for the check and check_existence rectangles. I don't think that it's useful to repeat it over and over. I think they could have a different tool tip, maybe related to the maining of the elements (see also the comment below).